### PR TITLE
Test LogsDB backward compatibility

### PR DESCRIPTION
**Pull request by salvatore-campagna**
Test LogsDB backward compatibility with a rolling upgrade and full cluster restart.
We try to start indexing logs using a `standard` index, then we switch to a `LogsDB` index.
We also improve the existing test which switches between the two index modes, `standard`
and `logs`.

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
This is still not using LogsDB...I am investigating the issue.

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                return sb.toString();
+            }));
+            assertOK(bulkIndexResponse);
+            assertThat(entityAsMap(bulkIndexResponse).get("errors"), Matchers.is(false));
+
+            assertThat(
+                getSettings(client(), getWriteBackingIndex(client(), "logs-apache-production", 0)).get("index.mode"),
+                Matchers.nullValue()
+            );
+            assertThat(
```
Comment (martijnvg) [inline qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java:172]: maybe for completeness also check whether synthetic source is enabled and index sorting is set? And also in the rolling upgrade test?

**Review by martijnvg — APPROVED** [review]
Left one suggestion otherwise LGTM

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java (hunk 1 above)
Comment (salvatore-campagna) [reply qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java:172]: The sorting if not explicitly applied is not part of the settings...I will test that in a different way as we discussed in Slack.

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+                },
+                "ip.address": {
+                  "type": "ip"
+                }
+              }
+            }
+          }
+        }""";
+
+    public void testLogsIndexing() throws IOException {
```
Comment (lkts) [inline qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java:122]: nit: maybe something like `testSwitchToLogsModeAndIndexData`?

Comment (lkts) [reply qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java:122]: Also do we want to have a similar test but in the opposite order (create with logs template initially)?

File: qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeRollingUpgradeIT.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+                },
+                "ip.address": {
+                  "type": "ip"
+                }
+              }
+            }
+          }
+        }""";
+
+    public void testLogsIndexing() throws IOException {
```
Comment (lkts) [inline qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeRollingUpgradeIT.java:123]: Same question - do we want to test in the opposite order?

File: qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java (hunk 2 above)
Comment (lkts) [reply qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeFullClusterRestartIT.java:122]: Should we do some validation that previously index document is retrievable?

File: qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeRollingUpgradeIT.java (hunk 3 above)
Comment (salvatore-campagna) [reply qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeRollingUpgradeIT.java:126]: This is a test upgrading a version of Elasticsearch that does not have LogsDB to a version that has it...so I think we can't test LogsDB starting the upgrade process from a version that is missing the functionality. Setting the index mode would just fail.

**Pull request merged**